app/components/user-profile.js

```diff
@@ -0,0 +1,161 @@
+import Ember from 'ember';
+import DS from 'ember-data';
+
+const { computed, Component, inject } = Ember;
+const { PromiseObject } = DS;
+const { service } = inject;
+
+export default Component.extend({
+  store: service(),
+
+  roles: computed({
+    get() {
+      const store = this.get('store');
+      return store.findAll('user-role');
+    }
+  }).readOnly(),
+
+  secondaryCohorts: computed('user', {
+    get() {
+      const user = this.get('model');
```

Comment:
It seems weird to keep calling the user the 'model' why not pass it in as user?  I'm ok merging like this, just a question.